cnf-tests/testsuites/e2esuite/gatekeeper/gatekeeper.go

```diff
@@ -55,616 +50,702 @@ var _ = Describe("gatekeeper", func() {
-		Expect(err).NotTo(HaveOccurred())
-		err = client.Get(context.Background(), includedNamespacedObjectKey, includedNamespacedObject)
-		Expect(err).NotTo(HaveOccurred())
+			value, ok = labels["cluster-selected"]
+			Expect(ok).To(Equal(true))
+			Expect(value).To(Equal("true"))
+
+			_, ok = labels["namespace-selected"]
+			Expect(ok).To(Equal(false))
+
+			By("Asserting that excludedNamespaced object mutations were applied")
+			excludedNamespacedObjectKey, err := k8sClient.ObjectKeyFromObject(excludedNamespacedObject)
+			Expect(err).NotTo(HaveOccurred())
+			err = client.Get(context.Background(), excludedNamespacedObjectKey, excludedNamespacedObject)
+			Expect(err).NotTo(HaveOccurred())
+
+			labels = excludedNamespacedObject.GetLabels()
+
+			value, ok = labels["all-selected"]
+			Expect(ok).To(Equal(true))
+			Expect(value).To(Equal("true"))
+
+			_, ok = labels["namespace-included"]
+			Expect(ok).To(Equal(false))
+
+			_, ok = labels["cluster-selected"]
+			Expect(ok).To(Equal(false))
+
+			_, ok = labels["namespace-selected"]
+			Expect(ok).To(Equal(false))
+
+			By("Asserting that includedNamespaced object mutations were applied")
+			includedNamespacedObjectKey, err := k8sClient.ObjectKeyFromObject(includedNamespacedObject)
+			Expect(err).NotTo(HaveOccurred())
+			err = client.Get(context.Background(), includedNamespacedObjectKey, includedNamespacedObject)
+			Expect(err).NotTo(HaveOccurred())
+
+			labels = includedNamespacedObject.GetLabels()
+
+			_, ok = labels["all-selected"]
+			Expect(ok).To(Equal(false))
+
+			value, ok = labels["namespace-included"]
+			Expect(ok).To(Equal(true))
+			Expect(value).To(Equal("true"))
+
+			_, ok = labels["cluster-selected"]
+			Expect(ok).To(Equal(false))
+
+			value, ok = labels["namespace-selected"]
+			Expect(ok).To(Equal(true))
+			Expect(value).To(Equal("true"))
+		})
+	})
+
+	Context("operator", func() {
+
+		It("should be able to select mutation namespaces", func() {
+			var err error
+
+			// Add namespace selector to operator config
+
+			// Create an all pod mutation
+			podMutation := &gkv1alpha.AssignMetadata{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "all-pod-mutation",
+				},
+				Spec: gkv1alpha.AssignMetadataSpec{
+					Match: gkv1alpha.Match{
+						Scope:      apiextensionsv1beta1.NamespaceScoped,
+						Namespaces: []string{TestingNamespace},
+						Kinds: []gkv1alpha.Kinds{
+							{
+								APIGroups: []string{""},
+								Kinds:     []string{"Pod"},
+							},
+						},
+					},
+					Location: "metadata.labels.mutated",
+					Parameters: gkv1alpha.MetadataParameters{
+						Assign: runtime.RawExtension{
+							Raw: []byte(fmt.Sprintf("{\"value\":\"true\"}")),
+						},
+					},
+				},
+			}
+			err = client.Create(context.Background(), podMutation)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create namespaces
+			By("Creating the test namespaces")
+			selectedTestNamespace := &corev1.Namespace{
```

Comment:
Can you please update the ns names to refelect that one's resources will be mutated?
Maybe:
selectedTestNamespace -> mutationEnabledTestNamespace
testNamespace - > mutationDisabledTestNamespace